lib/index.d.ts

```diff
@@ -0,0 +1,528 @@
+    client: IClient
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    maxAge: number
+  }
+
+  export interface IDeviceFlowHandle {
+    poll() : Promise<TokenSet>
+    expired() : boolean
+    expires_at: number
+    client: IClient
+    maxAge: number
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    interval: number
+    user_code: string
+    device_code: string
+    verification_uri: string
+    verification_uri_complete: string
+    expires_in: number
+  }
+
+  export class DeviceFlowHandle implements IDeviceFlowHandle {
+    constructor (params : IDeviceFlowHandleParams)
+    poll() : Promise<TokenSet>
+    expired() : boolean
+    // tslint:disable-next-line:variable-name
+    expires_at: number
+    client: IClient
+    maxAge: number
+    exchangeBody: object
+    clientAssertionPayload: object
+    response: string
+    interval: number
+    // tslint:disable-next-line:variable-name
+    user_code: string
+    // tslint:disable-next-line:variable-name
+    device_code: string
+    // tslint:disable-next-line:variable-name
+    verification_uri: string
+    // tslint:disable-next-line:variable-name
+    verification_uri_complete: string
+    // tslint:disable-next-line:variable-name
+    expires_in: number
+  }
+
+  export interface IIssuerMetadata {
+    issuer: string
+    authorization_endpoint: string
+    token_endpoint: string
+    jwks_uri: string
+    userinfo_endpoint: string
+    revocation_endpoint: string
+    end_session_endpoint: string
+    registration_endpoint: string
+    token_endpoint_auth_methods_supported: string
+    token_endpoint_auth_signing_alg_values_supported: string
+    introspection_endpoint_auth_methods_supported: string
+    introspection_endpoint_auth_signing_alg_values_supported: string
+    revocation_endpoint_auth_methods_supported: string
+    revocation_endpoint_auth_signing_alg_values_supported: string
+    request_object_signing_alg_values_supported: string
+    mtls_endpoint_aliases: IMetadataMtlsEndpointAliases
+
+    [key: string]: any
+  }
+
+  export interface IMetadataMtlsEndpointAliases {
+    token_endpoint: string
+    userinfo_endpoint: string
+    revocation_endpoint: string
+    introspection_endpoint: string
+  }
+
+  /**
+   * Encapsulates a discovered or instantiated OpenID Connect Issuer (Issuer), Identity Provider (IdP),
+   * Authorization Server (AS) and its metadata.
+   */
+  export interface IIssuer {
+    /**
+     * Returns the <Client> class tied to this issuer.
+     */
+    Client: { new (metadata: IClientMetadata, jwks?: object) }
+
+    /**
+     * Returns metadata from the issuer's discovery document.
+     */
+    metadata?: IIssuerMetadata
+
+    /**
+     * Returns the issuer's jwks_uri keys as a @panva/jose parsed JWKS.Keystore.
+     * @param forceReload
+     */
+    keystore (forceReload?: boolean): Promise<JWKS.KeyStore>
+
+    [key: string]: any
```

Comment:
As above, should an `Issuer` allow additional types like this?

Reply:
My linter fails if it's not included since it's implemented in `IIssuer`